saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/web/binding/DefaultSaml2MessageBindingResolver.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2002-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.saml2.provider.service.web.binding;
+
+import java.util.Collection;
+import java.util.List;
+
+import jakarta.servlet.http.HttpServletRequest;
+
+import org.springframework.security.saml2.provider.service.registration.Saml2MessageBinding;
+import org.springframework.util.Assert;
+
+public class DefaultSaml2MessageBindingResolver implements Saml2MessageBindingResolver {
```

Comment:
It doesn't seem likely enough to switch out binding resolution behavior to introduce a new interface. If you feel strongly about introducing this configurability, please open a separate ticket so we can discuss.

For this PR, please don't introduce new interfaces or public classes.

Reply:
I agree that it is not likely switching out binding resolution. Maybe I misunderstood your prior comment about preferring to place behavior in components that implement interfaces (https://github.com/spring-projects/spring-security/pull/11287#discussion_r883967313). Would it be preferred to have this logic in a single utility class or do you have other suggestions?

Reply:
@emopti-jrufer sorry, I can see how that wasn't clear. I don't think there's much value in an abstraction given that it would require changing or adding to the public API. 

I believe the result is that the behavior gets copied between classes.

Reply:
Yes, so long as references to `HttpServletRequest` stay in the `web` package and the public API is not affected, then a package-private abstraction should be fine.